### PR TITLE
Elaborate HSVColor interface

### LINKED ISSUE #507: HSVColor methods
- [x] `operator ==`
- [x] `hashCode`
- [x] `toString()`
- [x] `lerp()`

### PR DESCRIPTION
**Pull request by abarth**
Fixes #507

### CODE REVIEW

File: packages/flutter/lib/src/painting/colors.dart [hunk 1]
```diff
@@ -71,10 +75,52 @@ class HSVColor {
...
+    );
+  }
+
+  bool operator ==(dynamic other) {
+    if (identical(this, other))
+      return true;
+    if (other is! HSVColor)
+      return false;
+    final HSVColor typedOther = other;
+    return typedOther.alpha == alpha
```
Comment (Hixie) [inline packages/flutter/lib/src/painting/colors.dart]: We should be consistent about how we do && and || cascades. This and the || cascade in the patch earlier today or yesterday are the first times we've done it that way. They do look better in "return" cascades than what we normally do, though. Maybe we should say the rule is "in a return expression, if you have a sequence of subexpressions are delimited by && or ||, then put the logical operator under the return; in all other cases, when you line break at a logical operator, put it at the end of the line".

If you agree, file a bug to make the code consistent and to update the style guide.

File: packages/flutter/lib/src/painting/colors.dart [hunk 2]
```diff
@@ -2,45 +2,49 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-import 'dart:ui' show Color;
+import 'dart:ui' show Color, lerpDouble;
+
+HSVColor _scaleAlpha(HSVColor color, double factor) {
```
Comment (Hixie) [inline packages/flutter/lib/src/painting/colors.dart]: why not as a method of the class?

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/lib/src/painting/colors.dart (hunk 1 above)
Comment (abarth) [inline packages/flutter/lib/src/painting/colors.dart]: Let's fine a bug and figure this out.  I don't have much of an opinion.

File: packages/flutter/lib/src/painting/colors.dart (hunk 2 above)
Comment (abarth) [inline packages/flutter/lib/src/painting/colors.dart]: Fixed.

File: packages/flutter/lib/src/painting/colors.dart (hunk 1 above)
Comment (Hixie) [inline packages/flutter/lib/src/painting/colors.dart]: https://github.com/flutter/flutter/issues/519

**Pull request merged**